### PR TITLE
Create user functionality

### PR DESCRIPTION
**Pull request by Josephhaefling**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features
### Detailed Description
Created all user functionality
### Why is this change required? What problem does it solve?
These changes allow the user to search for rooms available on a chosen date. User can filter the results by room type and user can book room. User page updates after user chooses a room
### Was any previously written code changed?
no
### Files modified:
- [ ] index.html
- [ ] base.scss
- [ ] _mixins.scss
- [ ] _variables.scss
- [x ] index.js
- [ x] domUpdates.js
- [ ] User.js
- [ ] Hotel.js
- [ ] Bookings.js
- [ ] Manager.js
- [ ] images
- [ ] Other files:

### CODE REVIEW

**Comment by Josephhaefling** [comment]
@hannahhch 
@thatpamiam
@infamouskeyduster 
Hey y'all I am done with iteration 3. John Can you take a look and merge this form me?

File: src/Manager.js [hunk 1]
```diff
@@ -1,40 +1,34 @@
...
   getBookedRooms(bookingRepo) {
-    this.hotelRooms.forEach(room => {
-      this.todaysBookings.forEach(booking => {
-        booking.roomNumber === room.number ? this.bookedRooms.push(room) : console.log();
-      })
-    })
+    let todaysBookings = this.getBookingStatus(bookingRepo)
+      return todaysBookings.map(booking => {
+        return this.hotelRooms.filter(room => room.number === booking.roomNumber)
+    }).flat()
```
Comment (infamouskeyduster) [inline src/Manager.js:19]: I like you use of .flat() array prototype here! That can come in super handy to make a single array out of nested arrays!

File: src/Manager.js [hunk 2]
```diff
@@ -1,40 +1,34 @@
...
+    return Math.round((this.getVacantRooms(roomRepo, bookingRepo).length / roomRepo.length) * 100)
   }
 
-  getTotalRevenue() {
-    let revenueSum = this.bookedRooms.reduce((totalRev, room) => {
+  getTotalRevenue(bookingRepo) {
+    let revenueSum = this.getBookedRooms(bookingRepo).reduce((totalRev, room) => {
       totalRev += room.costPerNight
       return totalRev
     }, 0)
```
Comment (infamouskeyduster) [inline src/Manager.js:35]: All of your methods look/seem sound. Looks like your logic is good to return the specific things that you are looking for!

File: src/index.js [hunk 3]
```diff
@@ -73,16 +73,20 @@ let createUsers = (userLoginRepo) => {
 let createHotel = (date) => {
   let skeletInn = new Hotel(date, roomRepo)
   createManager(todaysDate, skeletInn, roomRepo)
+  getAvailableRooms(skeletInn)
 }
 
 let createManager = (currentDate, hotel, roomRepo) => {
   let lucyPhurr = new Manager(currentDate, hotel, roomRepo)
```
Comment (infamouskeyduster) [inline src/index.js:86]: What a wild name for your instantiated manager!

File: test/Manager-test.js [hunk 4]
```diff
@@ -42,23 +50,44 @@ describe('Manager', () => {
...
   it('should get the percentage of available rooms', () => {
-    expect(lucyPhurr.getPercentageOfRoomsAvailable()).to.equal(100)
+    createBookingRepo()
+    // let bookedRooms = lucyPhurr.getBookedRooms(bookingRepo)
+    // let getVacantRooms = lucyPhurr.getVacantRooms(roomRepo, bookingRepo)
+    expect(lucyPhurr.getPercentageOfRoomsAvailable(roomRepo, bookingRepo)).to.equal(43)
   })
-  it('should get the total revenue for today.', () => {
-    expect(lucyPhurr.getTotalRevenue()).to.equal(0)
+  it.only('should get the total revenue for today.', () => {
```
Comment (infamouskeyduster) [inline test/Manager-test.js]: Make sure to remove this it.only before proceeding with any more testing of this class! That can really mess things up of it is forgotten about!

**Comment by infamouskeyduster** [comment]
Alright! I looked over your code… things look well designated, it looks like your classes are returning the desired properties correctly! I did leave a comment on your manager test (I believe), there is an it.only that was left in your testing suite. Just remember to remove this before continuing with testing. This concludes my review! I'm going to merge it for you, but my code review may require addressing before merging.

**Comment by infamouskeyduster** [comment]
Looks like my review of your it.only was done on code that is now outdated, so the merge went through without any issue!

**Pull request merged**